### PR TITLE
Rebase stm32.js

### PR DESCRIPTION
**Pull request by haslinghuis**
Rebase of stm32.js:

- [x] fix lexical scoping
- [x] change strings into template literals
- [x] remove type coercion
- [x] change (most) variables from kebab_case to camelCase 

Only one file to go (usbstm32dfu.js)

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by Asizon — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]
Good cleaning! I have not tested it, but seems good to me...

**Review by ctzsnooze — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/protocols/stm32.js
Comment (mikeller) [inline src/js/protocols/stm32.js:322]: Looking at `src/js/serial.js`, `callback` seems to be optional, so we can probably just omit this.

File: src/js/protocols/stm32.js
Comment (mikeller) [inline src/js/protocols/stm32.js:426]: Did you want to fix `first_array` as well?

File: src/js/protocols/stm32.js
Comment (mikeller) [inline src/js/protocols/stm32.js:289]: Did you want to fix `n_bytes` as well?

File: src/js/protocols/stm32.js
Comment (mikeller) [inline src/js/protocols/stm32.js:665]: Not sure why this is capitalised?

File: src/js/protocols/stm32.js
Comment (haslinghuis) [reply src/js/protocols/stm32.js]: Reduced.

File: src/js/protocols/stm32.js
Comment (haslinghuis) [reply src/js/protocols/stm32.js]: Renamed

File: src/js/protocols/stm32.js
Comment (haslinghuis) [reply src/js/protocols/stm32.js]: Renamed

File: src/js/protocols/stm32.js
Comment (haslinghuis) [reply src/js/protocols/stm32.js]: Yes, as original, fixed now.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
Do I need to do anything here to get it approved?

**Review by McGiverGim — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
